### DISCUSSION TITLE
Filtering users by DeviceID in Z-push 2.7.0

### DISCUSSION THREAD

**Discussion by imayma** [Z-Hub/Z-Push#29] [Q&A] [answered]
Hello,

I recently installed Z-push 2.7.0 on Ubuntu server 22.04 using Zimbra Backend. 

The thnig I want now is filtering the authorized users only to connect to it. I tried by adding this to /etc/z-push/z-push-config.php : 

define('USE_DEVICEID', true);
$allowedDeviceIDs = array(
    'DeviceID1',
    'DeviceID2',
    // Ajoutez les ID des périphériques autorisés ici
);

// Vérifie si l'ID du périphérique est autorisé
function isAllowedDeviceID($deviceID) {
    global $allowedDeviceIDs;
    return in_array($deviceID, $allowedDeviceIDs);
}

But it doesn't work at all.

Is there any possibility to protect the z-push of unwanted connection by allowing Device ID or for example ?

**Comment by imayma** [comment, marked as answer]
I found myself an answer using this article : 

https://journeymblog.wordpress.com/2017/11/18/zimbra-z-push-part-3-user-restriction/

And by adapting the code to my configuration

**Comment by matidau** [comment]
Hope you don't mind me re-opening the discussion, closed discussions make them a little hard to find.

Quick question are you using the Zimbra backend, or are you using another one and modified the code from the blog for your purposes?

In general I would recommend to use the fail2ban and the LDAP backend (though this might not be compatible with the Zimbra backend).

**Comment by matidau** [comment]
And I'm closing it now, as Github has made a change that shows closed discussions by default.
